pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -743,6 +743,16 @@ func permissionForKeyValues(key string, values []string) *policyproto.Permission
 				},
 			}, nil
 		}
+	case isKeyBinary(key) && !attributesEnforcedInPlugin(key):
+		// Split key of format a[b] to [a, b].
+		parts := strings.Split(strings.TrimSuffix(key, "]"), "[")
+		converter = func(v string) (*policyproto.Permission, error) {
+			return &policyproto.Permission{
+				Rule: &policyproto.Permission_Metadata{
+					Metadata: generateMetadataListMatcher(parts[0], parts[1:], v),
```

Comment:
You are assuming this it's always a list matcher in the API, this makes it very hard for any future update to the API and also it's locked with one specific use case.

Reply:
+1. This is too general. It basically says all properties with the format of a[b] will be handled in listMatcher. I am concerned that it will cause unexpected behavior and instability of the code.

Reply:
To be more specific on the risk part. This change applies to all the cases that properties "key" is NOT in the supported list (https://preliminary.istio.io/docs/reference/config/authorization/constraints-and-properties/). We have clearly documented that the unsupported keys will be ignored silently. This change completely changes the behavior. 

Consider the case that a user has a typo in his policy file (e.g., source.ip => sources.ip). Instead of being silently ignored, this wrong key "sources.ip" will be interpreted into some metadata matching and cause unexpected behavior. 

Another problem is, all the unknown keys are translated into listmatcher. This is problematic because we do not know the value types. What if the values are integer, or boolean, etc? What will happen in that case? Will it cause proxy to crash or any unexpected runtime error?